frontend/wallet/src/render/components/TransactionsView.re

```diff
@@ -37,78 +37,61 @@ module Styles = {
-    }),
-    TransactionCell.Transaction.Payment(
-      {
-        from: otherKey,
-        to_: List.head(myWallets) |> Option.getExn,
-        amount: Int64.of_int(1540),
-        fee: Int64.of_int(10),
-        memo: Some("Remittance payment"),
-        includedAt: Some(Js.Date.fromString("16 Apr 2019 21:46:00 PST")),
-        submittedAt: Js.Date.fromString("15 Apr 2019 21:46:00 PST"),
-      },
-      {
-        status: ConsensusState.Status.Submitted,
-        estimatedPercentConfirmed: 0.0,
-      },
-    ),
-    TransactionCell.Transaction.Payment(
-      {
-        from: otherKey,
-        to_: List.head(myWallets) |> Option.getExn,
-        amount: Int64.of_int(2415),
-        fee: Int64.of_int(10),
-        memo: Some("Order #: 2347B342"),
-        includedAt: Some(Js.Date.fromString("16 Apr 2019 21:46:00 PST")),
-        submittedAt: Js.Date.fromString("15 Apr 2019 21:46:00 PST"),
-      },
-      {status: ConsensusState.Status.Failed, estimatedPercentConfirmed: 0.0},
-    ),
-  ]);
+module Transactions = [%graphql
+  {|
+    query transactions($publicKey: String!) {
+      blocks(filter: { relatedTo: $publicKey }) {
+        nodes {
+          creator @bsDecoder(fn: "Apollo.Decoders.publicKey")
+          protocolState {
+            blockchainState {
+              date @bsDecoder(fn: "Apollo.Decoders.date")
+            }
+          }
+          transactions {
+            userCommands {
+              to_: to @bsDecoder(fn: "Apollo.Decoders.publicKey")
+              from @bsDecoder(fn: "Apollo.Decoders.publicKey")
+              amount @bsDecoder(fn: "Apollo.Decoders.int64")
+              fee @bsDecoder(fn: "Apollo.Decoders.int64")
+              memo
+              isDelegation
+              date @bsDecoder(fn: "Apollo.Decoders.date")
+            }
+            feeTransfer {
+              recipient @bsDecoder(fn: "Apollo.Decoders.publicKey")
+              amount @bsDecoder(fn: "Apollo.Decoders.int64")
+            }
+            coinbase @bsDecoder(fn: "Apollo.Decoders.int64")
+          }
+        }
+      }
+    }
+  |}
+];
+module TransactionsQuery = ReasonApollo.CreateQuery(Transactions);
+
+let extractTransactions: Js.t('a) => array(TransactionCell.Transaction.t) = data => {
+  data##blocks##nodes
+  |> Array.map(~f=block => {
+       open TransactionCell.Transaction;
+       let userCommands =
+         block##transactions##userCommands
+         |> Array.map(~f=userCommand => UserCommand(userCommand));
+       let blockReward =
+         BlockReward({
+           date: block##protocolState##blockchainState##date,
+           creator: block##creator,
+           coinbase: block##transactions##coinbase,
+           feeTransfers: block##transactions##feeTransfer,
+         });
+       Array.append(userCommands, [|blockReward|]);
+     }) |> Array.concatenate;
+};
 
 [@react.component]
-let make = () =>
+let make = () => {
+  let transactionsQueryThing = Transactions.make(~publicKey="123", ());
```

Comment:
maybe just `transactionsQuery`? haha
And this pubkey should be the active wallet, right?